src/middlewared/middlewared/plugins/auth.py

```diff
@@ -591,17 +591,7 @@ def generate_base32_secret(self):
         return pyotp.random_base32()
 
 
-async def check_permission(middleware, app):
-    """
-    Authenticates connections coming from loopback and from
-    root user.
-    """
-    sock = app.request.transport.get_extra_info('socket')
-    if sock.family == socket.AF_UNIX:
-        # Unix socket is only allowed for root
-        AuthService.session_manager.login(app, UnixSocketSessionManagerCredentials())
-        return
-
+def authenticate_connection(app):
     remote_addr, remote_port = get_remote_addr_port(app)
```

Comment:
If we want to eliminate all nanodelays and achieve super turbo boost performance, then we don't need to perform this in a thread, only the `get_peer_process` bit.

Reply:
It's not about "achieve super turbo boost performance", It's about doing what's right. Doing file IO in the main event loop is incorrect but you already know this, however, so I appreciate constructive reviews instead of reviews filled with sarcasm.

Also, if you look at the code, you'll see that we call `get_remote_addr_port` which can call `get_peer_process` in the `else` branch. Then we call `get_peer_process` again potentially.

Reply:
I think the whole point of doing things in a thread is precisely because we don't want to block for I/O in the main event loop.